packages/evm/src/fungible.ts

```diff
@@ -0,0 +1,185 @@
+import type { Domainlike, EvmResource } from '@buildwithsygma/core';
+import { SecurityModel, Config, FeeHandlerType } from '@buildwithsygma/core';
+import { Bridge__factory, ERC20__factory } from '@buildwithsygma/sygma-contracts';
+import { Web3Provider } from '@ethersproject/providers';
+import { BigNumber, constants, providers, utils, type PopulatedTransaction } from 'ethers';
+import type { Eip1193Provider, EvmFee, TransactionRequest } from 'types.js';
+
+import { BaseTransfer } from './base-transfer.js';
+import { BasicFeeCalculator } from './fee/BasicFee.js';
+import { PercentageFeeCalculator } from './fee/PercentageFee.js';
+import { getFeeInformation } from './fee/getFeeInformation.js';
+import { approve, getERC20Allowance } from './utils/approveAndCheckFns.js';
+import { erc20Transfer } from './utils/depositFns.js';
+import { createTransactionRequest } from './utils/transaction.js';
+
+type EvmFungibleTransferRequest = {
+  source: Domainlike;
+  destination: Domainlike;
+  sourceAddress: string;
+  sourceNetworkProvider: Eip1193Provider;
+  resource: string | EvmResource;
+  amount: bigint;
+  destinationAddress: string;
+  securityModel?: SecurityModel;
+};
+
+export async function createEvmFungibleAssetTransfer(
+  params: EvmFungibleTransferRequest,
+): Promise<EvmFungibleAssetTransfer> {
+  const config = new Config();
+  await config.init(process.env.SYGMA_ENV);
+
+  const transfer = new EvmFungibleAssetTransfer(params, config);
+
+  const isValid = await transfer.isValidTransfer();
+  if (!isValid)
+    throw new Error('Handler not registered, please check if this is a valid bridge route.');
+
+  const originalFee = await transfer.getFee();
+  //in case of percentage fee handler, we are calculating what amount + fee will result int user inputed amount
+  //in case of fixed(basic) fee handler, fee is taken from native token
+  if (originalFee.type === FeeHandlerType.PERCENTAGE) {
+    let _amount = transfer.amount;
+    const minFee = originalFee.minFee!;
+    const maxFee = originalFee.maxFee!;
+    const percentage = originalFee.percentage!;
+    const userInputAmount = BigNumber.from(transfer.amount);
+
+    //calculate amount without fee (percentage)
+    const feelessAmount = userInputAmount
+      .mul(constants.WeiPerEther)
+      .div(utils.parseEther(String(1 + percentage)));
+
+    const calculatedFee = userInputAmount.sub(feelessAmount);
+    _amount = feelessAmount.toBigInt();
+    //if calculated percentage fee is less than lower fee bound, substract lower bound from user input. If lower bound is 0, bound is ignored
+    if (calculatedFee.lt(minFee) && minFee > 0) {
+      _amount = transfer.amount - minFee;
+    }
+    //if calculated percentage fee is more than upper fee bound, substract upper bound from user input. If upper bound is 0, bound is ignored
+    if (calculatedFee.gt(maxFee) && maxFee > 0) {
+      _amount = transfer.amount - maxFee;
+    }
+    transfer.setAmount(_amount);
```

Comment:
Probably we could move it to a separate private method like "calculatePercantageFee"

Reply:
Also potentially we can move it to the EvmFungibleAssetTransfer class

Reply:
thanks, this method can not be a private member as it needs to be called outside the context of the class therefore ive added another method for this